docs/data/toolpad/core/components/persistent-state/LocalStorageStateError.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { Codec, useLocalStorageState } from '@toolpad/core';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+
+// preview-start
+const CODEC_JSON: Codec<unknown> = {
+  parse: (value) => {
+    try {
+      return JSON.parse(value);
+    } catch {
+      return { _error: 'parse failed' };
+    }
+  },
+  stringify: (value) => JSON.stringify(value),
+};
+// preview-end
+
+export default function LocalStorageStateError() {
+  const [value, setValue] = useLocalStorageState('error-value', null, {
+    codec: CODEC_JSON,
+  });
+  return (
+    <Stack spacing={2} alignItems="center">
+      <pre>value: {JSON.stringify(value)}</pre>
+
+      <Stack direction="row" spacing={2}>
+        <Button onClick={() => setValue({ foo: Math.random() })}>Set JSON</Button>
+        <Button
+          onClick={() => window.localStorage.setItem('error-value', 'invalid json')}
```

Comment:
If using local storage directly I would expect this to just work, should this not be using `setValue` from the hook? (but I might be misunderstanding something)

Reply:
you can not set invalid json with setValue because it takes care of serialization